### PR TITLE
Update doc for starter project

### PR DESCRIPTION
**Pull request by adisky**
**What type of PR is this?**
 /kind documentation
 [skip ci]

**What does does this PR do / why we need it**:
Update devfile reference index for starter projects

**Which issue(s) this PR fixes**:

Fixes #?

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [x] Documentation 

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: docs/file-reference/index.md [hunk 1]
```diff
@@ -297,6 +319,59 @@ projects:
...
+      branch: master
+      sparseCheckoutDir: /getting-started/ # optional checkout dir
+      startPoint: 1.7.0.Final # tag or commit to start at
+```   
+
+
+> Pulling from a GitHub location
+
+```yaml
+projects:
```
Comment (cdrage) [inline docs/file-reference/index.md:341]: if we removed `projects` shouldn't we remove these parts too? Bit confused. Thought we were removing `projects` too haha

File: docs/file-reference/index.md [hunk 2]
```diff
@@ -43,14 +45,14 @@ commands:
...
-| components    | array of [componentObject](#component-object) | no       | List of components to be used within your development environment                 |
-| commands      | array of [commandObject](#commandobject)      | no       | List of commands to be executed                                                   |
-
+| Key             | Type                                                    | Required | Description                                                             |
+|-----------------|---------------------------------------------------------|----------|-------------------------------------------------------------------------|
+| schemaVersion   | string                                                  | yes      | Schema version of devfile                                               |
+| metadata        | [metadataObject](#metadataobject)                       | no       | Metadata information that describes the project                         |
+| starterProjects | array of [starterProjectObject](#starterproject-object) | no       | List of starter projects that can be used to bootstrap new projects.    |
```
Comment (cdrage) [inline docs/file-reference/index.md:52]: remove period end of `List of starter projects that can be used to bootstrap new projects` sentence

File: docs/file-reference/index.md [hunk 3]
```diff
@@ -43,14 +45,14 @@ commands:
...
+| projects        | array of [projectObject](#project-object)               | no       | List of projects that devfile use when creating development environment |
```
Comment (cdrage) [inline docs/file-reference/index.md:53]: you changed the description for some reason? It should be saying: `List of projects that devfile will use when creating your development environment`

File: docs/file-reference/index.md [hunk 4]
```diff
@@ -75,6 +77,25 @@ metadata:
...
 
+## starter projects
+
+> Example using a [Quarkus quickstart project](https://github.com/quarkusio/quarkus-quickstarts)
+
+```yaml
+starterProjects:
+  - name: quarkus-quickstart
+    git:
+      location: https://github.com/quarkusio/quarkus-quickstarts
```
Comment (cdrage) [inline docs/file-reference/index.md:88]: you're missing adding the `description` key to this example

File: docs/file-reference/index.md [hunk 5]
```diff
@@ -297,6 +319,59 @@ projects:
...
+```
+
+Each project may contain three different objects, `git`, `github` or `zip`.
+
+
+
+| Key       | Type                          | Description                                                                        |
+|-----------|-------------------------------|------------------------------------------------------------------------------------|
+| name      | string                        | Name of your devfile                                                               |
+| description | string                      | description of a starter project
```
Comment (cdrage) [inline docs/file-reference/index.md:369]: you're forgetting a `|` at the end of this

need to capitalize Description

File: docs/file-reference/index.md [hunk 6]
```diff
@@ -297,6 +319,59 @@ projects:
...
+      location: https://github.com/odo-devfiles/nodejs-ex.git
+      branch: master
+      sparseCheckoutDir: /app/ # optional checkout dir
+      startPoint: 1.0.0 # tag or commit to start at
+```
+
+> Pulling from a zip 
+
+```yaml
+projects:
```
Comment (cdrage) [inline docs/file-reference/index.md:354]: these aren't using `starterProjects` this says `projects`

File: docs/file-reference/index.md (hunk 2 above)
Comment (adisky) [reply docs/file-reference/index.md:52]: ya we are removing, I raised PR before the slack discussion, still working on it

**Comment by girishramnani** [comment]
Unit test failing on this

File: docs/file-reference/index.md (hunk 4 above)
Comment (adisky) [reply docs/file-reference/index.md:86]: added

File: docs/file-reference/index.md (hunk 1 above)
Comment (adisky) [reply docs/file-reference/index.md]: removed projects

File: docs/file-reference/index.md (hunk 5 above)
Comment (adisky) [reply docs/file-reference/index.md]: done.

File: docs/file-reference/index.md (hunk 3 above)
Comment (adisky) [reply docs/file-reference/index.md]: not applicable now.

**Comment by girishramnani** [comment]
will wait for your approval on this @cdrage

**Pull request merged**